src/compute-client/src/types/sinks.proto

```diff
@@ -21,7 +21,7 @@ message ProtoComputeSinkDesc {
     mz_repr.global_id.ProtoGlobalId from = 1;
     mz_repr.relation_and_scalar.ProtoRelationDesc from_desc = 2;
     ProtoComputeSinkConnection connection = 3;
-    ProtoSinkAsOf as_of = 4;
```

Comment:
Are we sure that's never been written down? In the future, we should instead reserve 4 and add a new number for `with_snapshot`.

Reply:
> Are we sure that's never been written down?

Yes, if you remove the `RustType` impl for `ComputeSinkDesk` only the `RustType` impl for `DataflowDescription` complains. And if you remove that only the grpc client complains.

> In the future, we should instead reserve 4 and add a new number for with_snapshot.

I'm wary of forcing ourselves to stay backward compatible when we don't need to. It would be fine in this case, but if we added a field with a more complex type, we would have to make that an `Option` to stay backward compatible,  in which case the surrounding code would be burdened with having to somehow handle `None` values.